code/modules/atmospherics/machinery/components/unary_devices/vent_scrubber.dm

```diff
@@ -135,7 +140,24 @@
 	check_turfs()
 	. = ..()
 
-/obj/machinery/atmospherics/components/unary/vent_scrubber/process_atmos()
+/obj/machinery/atmospherics/components/unary/vent_scrubber/should_atmos_process(datum/gas_mixture/air, exposed_temperature)
+	if(welded || !is_operational)
+		return FALSE
+	if(!nodes[1] || !on || !filter_types)
+		on = FALSE
+		return FALSE
+
+	var/list/changed_gas = air?.gases
+
+	if(!changed_gas)
+		return FALSE
+
+	if(scrubbing == SIPHONING || length(filter_types & changed_gas))
+		return TRUE
+
+	return FALSE
+
+/obj/machinery/atmospherics/components/unary/vent_scrubber/atmos_expose(datum/gas_mixture/air, exposed_temperature)
 	if(welded || !is_operational)
 		return FALSE
```

Comment:
This on its own is not enough. `should_atmos_process` will only be called automatically if the scrubber's turf's mix changes. You will need to manually call it when anything else that might trigger a valid scrub changes

Reply:
oh yeah i forgot air has lower wait than machines. i still dont like having it in two subsystem processes though when its only ever used here so ill add a cooldown